awslabs/aws-servicebroker
# Enabling Service Instance Sharing

## Overview

Brief description of what this PR does, and why it is needed (use case)?
See https://docs.cloudfoundry.org/services/enable-sharing.html 

CloudFoundry allows for services to be shared across orgs and spaces -- however this requires that the service broker is explicitly enabling service instance sharing by setting a flag shareable: true. At this time, the AWS ServiceBroker does not pass this variable.  This PR will change that.

This has backwards compatibility since it is technically without the value, it is defaulted as false.  User must include in the template `shareable: true` and the service broker via CloudFoundry handles the rest.



## Related Issues

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #249

## Testing

How did you validate the changes in this PR? If there are unit tests included describe what they test

- No unit tests.
- Validated in lab & used in our current cloud foundry environment without fail.

### Notes

https://docs.cloudfoundry.org/services/enable-sharing.html 

## Testing Instructions

 How to test this PR Start after checking out this branch (bulleted)

* Add `shareable: true`  to a template
* In CloudFoundry's `ccdb`, confirm that the `services` table has in the `extras` field `{shareable: true}`
* Confirm in cf CLI that you can now perform `cf share-service <the-service-instance> -s <space-to-share-it>`

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
